components/script/dom/namednodemap.rs

```diff
@@ -99,8 +99,15 @@ impl NamedNodeMapMethods for NamedNodeMap {
 
     // https://heycam.github.io/webidl/#dfn-supported-property-names
     fn SupportedPropertyNames(&self) -> Vec<DOMString> {
-        self.owner.attrs().iter().map(|attr| {
-            DOMString::from(&**attr.name()) // FIXME(ajeffrey): Convert directly from &Atom to DOMString
-        }).collect()
+        let mut string_vec = vec!();
+        for attr in self.owner.attrs().iter() {
+            // FIXME(ajeffrey): Convert directly from &Atom to DOMString
+            let s = DOMString::from(&**attr.name());
+            if self.owner.html_element_in_html_document() && s != &*s.to_lowercase() {
```

Comment:
`to_lowercase` is almost always wrong in DOM code; try to write a test with uppercase characters outside A-Z that fails with this code.